### PR TITLE
Add the ability to switch languages + Spanish translations

### PR DESCRIPTION
**Pull request by mjgiarlo**
Ported from projecthydra/sufia#2766 and projecthydra/sufia#2772 -- thanks to @ejrivero and @jcoyne for this work!

screenshot from 2016-12-14 11-22-17

Click the Spanish flag to switch to using the Spanish locale.

Yes, the locales all over the place are kinda gross.

### CODE REVIEW

**Comment by jcoyne** [comment]
👍

**Comment by mjgiarlo** [comment]
@jcoyne how good does it look? ;)

**Comment by jcoyne** [comment]
@mjgiarlo Looks like there's a conflict now.  I didn't want to merge as a lot of this PR is my work.

**Comment by mjgiarlo** [comment]
@jcoyne Fixed the conflicts. Builds now running.

File: app/helpers/hyrax/hyrax_helper_behavior.rb [hunk 1]
```diff
@@ -14,6 +15,12 @@ module HyraxHelperBehavior
     include Hyrax::ChartsHelper
     include Hyrax::DashboardHelperBehavior
 
+    # Which translations are available for the user to select
+    # @return [Hash<String,String>] locale abbreviations as keys and flags as values
+    def available_translations
+      { 'en' => '🇬🇧', 'es' => '🇪🇸' }
```
Comment (anarchivist) [inline app/helpers/hyrax/hyrax_helper_behavior.rb]: [Flags should not represent languages](http://www.flagsarenotlanguages.com/blog/why-flags-do-not-represent-language/).

Comment (jpstroop) [inline app/helpers/hyrax/hyrax_helper_behavior.rb]: :+1: to :-1: on flags

Comment (jcoyne) [inline app/helpers/hyrax/hyrax_helper_behavior.rb]: This instead? https://github.com/projectblacklight/demo.projectblacklight.org/pull/2

Comment (anarchivist) [inline app/helpers/hyrax/hyrax_helper_behavior.rb]: I'd 👌 that, but maybe `English` and `Español` would be adequate?

Comment (mjgiarlo) [inline app/helpers/hyrax/hyrax_helper_behavior.rb]: I'll pull out the flags.

Comment (ggeisler) [inline app/helpers/hyrax/hyrax_helper_behavior.rb]: I'd consider this approach when there are only two languages:

bassi-veratti_collection

And when there are three or more, make it a dropdown, showing the current language as the label and the other available languages as the menu options.

File: app/views/shared/_locale_picker.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        <span class="sr-only"><%= t('hyrax.toolbar.language_switch') %></span>
+        <span title="<%= t('hyrax.toolbar.language_switch') %>"><%= available_translations[I18n.locale.to_s] %></span>
+        <b class="caret"></b>
+    </a>
+    <ul class="dropdown-menu" role="menu">
+        <li role="presentation" class="dropdown-header"><%= t('hyrax.toolbar.language_switch') %></li>
+        <li role="presentation" class="divider"></li>
+        <% available_translations.each do |language, label| %>
+            <li role="presentation" lang="<%= language %>">
+                <%= link_to label, "?locale=#{language}", class: 'dropdown-item', role: 'menuitem', taxindex: '-1' %>
```
Comment (mjgiarlo) [inline app/views/shared/_locale_picker.html.erb]: Typo. Pushing a change.

**Comment by mjgiarlo** [comment]
Traviiiiiiiiis. :red_circle:

**Comment by mjgiarlo** [comment]
This PR has been reviewed by a few folks and approved. I believe I've responded to all feedback thus far gathered. I am merging the PR, since this work has been around (in Sufia) since October, and I believe we've done due diligence. If folks have strong objections, hit me up and I will revert and revisit. Thanks, all.

**Pull request merged**